### PR TITLE
Refactor amount_usd calculation in paraswap.trades

### PR DESCRIPTION
**Pull request by Hosuke**
Linked issue: #6215 

### Contribution type
Please check the type of contribution this pull request is for:

- [ ] New spell(s)
- [x] Adding to existing spell lineage
- [ ] Bug fix

**Note:** You can safely discard any section below which doesn't apply based on selection above

---

### For new spell(s)
If you are building new spell(s), please provide the following information:

- **Spell name(s):** *schema.table_name*
- **Description:** [Detailed description of the new spell(s) and their purpose]
- **Who are the new spell(s) for?** [Internal team or general community]
- **How will the new spell(s) be used downstream?** [Description of downstream usage]
- **Implementation details:** [Information on how the spell(s) are implemented]
- **Test instructions:** [How to test the new spell(s)]
- **Related issue(s):** [Link to related issues, if any]

---

### For adding to existing spell lineage
If you are adding to an existing spell lineage, please provide the following information:

- **Description:** [Description of the changes made]

---

### For bug fixes
If you are fixing a bug, please provide the following information:

- **Description:** [Description of the bug fix]
- **Steps to reproduce:** [How to reproduce the bug]
- **Implementation details:** [Information on how the bug was fixed]
- **Test instructions:** [How to test the fix]
- **Related issue(s):** [Link to related issues, if any]

---

### Additional information
Please provide any additional information that might help us review your pull request:

- [Any additional information]

---

Thank you for your contribution!

### CODE REVIEW

**Comment by Hosuke** [comment]
Currently I keep avalanche_c implementation untouched.

File: dex/models/_projects/paraswap/avalanche_c/paraswap_v5_avalanche_c_trades.sql [hunk 1]
```diff
@@ -60,60 +60,6 @@ dexs as (
...
         {% endif %}
     {% endfor %}
-),
-
--- USDC.e AND USDT.e price are missing before 2022-10
-price_missed_previous AS (
-    WITH usdc_price AS (
-        SELECT minute, contract_address, decimals, symbol, price
-        FROM {{ source('prices', 'usd') }}
-        WHERE contract_address = 0xa7d7079b0fead91f3e65f86e8915cb59c1a4c664 -- USDC.e
```
Comment (Hosuke) [inline dex/models/_projects/paraswap/avalanche_c/paraswap_v5_avalanche_c_trades.sql:70]: Removed these parts after testing:
https://dune.com/queries/3848201?category=abstraction&namespace=paraswap&id=paraswap.trades

**Comment by Hosuke** [comment]
Should we keep the `post_hook` in sub spells?

File: dex/models/_projects/paraswap/arbitrum/paraswap_v5_arbitrum_trades.sql [hunk 2]
```diff
@@ -7,10 +7,10 @@
...
     incremental_strategy = 'merge',
     unique_key = ['block_date', 'blockchain', 'project', 'version', 'tx_hash', 'evt_index', 'trace_address'],
-    post_hook='{{ expose_spells(\'["arbitrum"]\',
-                                "project",
-                                "paraswap_v5",
-                                \'["springzh"]\') }}'
+    post_hook='{{ expose_spells(blockchains = \'["arbitrum"]\',
+                                spell_type = "project",
+                                spell_name = "paraswap_v5",
+                                contributors = \'["springzhang"]\') }}'
```
Comment (Hosuke) [inline dex/models/_projects/paraswap/arbitrum/paraswap_v5_arbitrum_trades.sql:13]: Hi @jeff-dude,
should I remove the `post_hook` in blockchain-version-level spells?

File: dex/models/_projects/paraswap/paraswap_trades.sql [hunk 3]
```diff
@@ -19,39 +19,72 @@ ref('paraswap_avalanche_c_trades')
 ,ref('paraswap_base_trades')
 ] %}
 
+WITH basic_trades AS (
```
Comment (Hosuke) [inline dex/models/_projects/paraswap/paraswap_trades.sql:22]: Hi @jeff-dude,
should I use `basic_trades` or simply `trades` here for the CTE naming?
As I may add similar CTE across the spells, so I think we may have an unified naming standard.

**Comment by Hosuke** [comment]
Updated.

**Comment by jeff-dude** [comment]
this is last remaining "leftover" PR that will eventually get replaced with new one. i'm going to close this, it'll be there in closed for historical purposes.

**Pull request closed without merging**